warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexConfig.java

```diff
@@ -0,0 +1,168 @@
+package datawave.ingest.data.config.ingest;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Container class for keeping track of configuration rules related to Whindex entries.
```

Comment:
A class-level comment should not contain this much detail about the internal variables of the class unless there is a good reason for doing so, such as explaining a complex process that is handled by the class. We already have documentation about the purpose of the internal members through the comments directly above them. A class-level summary should focus more on being a summary of the purpose of the class. It is often fine for this to be a single sentence for a simple class like this.